library/src/main/java/com/google/android/exoplayer/text/webvtt/WebvttParser.java

```diff
@@ -164,29 +182,32 @@ public final WebvttSubtitle parse(InputStream inputStream) throws IOException {
 
         try {
           if ("line".equals(name)) {
+            Pair<String, Alignment> lineMetadata = parseLinePositionAttributes(value);
+            value = lineMetadata.first;
+            lineAlignment = lineMetadata.second;
             if (value.endsWith("%")) {
               lineNum = parseIntPercentage(value);
-            } else if (value.matches(NON_NUMERIC_STRING)) {
-              Log.w(TAG, "Invalid line value: " + value);
             } else {
-              lineNum = Integer.parseInt(value);
+              // Following WebVTT spec, line number can be a negative number
+              int sign = 1;
```

Comment:
Integer.parseInt can parse the string including the negative sign, so you don't need to do this manually. You just need to make sure the ```value.matches(NON_NUMERIC_STRING)``` line doesn't trigger for negative numeric values.